### ISSUE TITLE
[MASSEMBLY-555] Need a better solution than packaging == pom for distribution modules whose output is an assembly

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#722]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** opened **[MASSEMBLY-555](https://issues.apache.org/jira/browse/MASSEMBLY-555?redirect=false)** and commented

It's common practice to use a purpose-built module to create a distribution assembly for a project. However, there is no packaging that accommodates this. We need one or more packagings added to the assembly plugin that will allow a proper non-pom packaging.

Currently, this is an issue if someone tries to create one assembly from a purpose-built module - say, for an examples aggregation - and then use it in another assembly via moduleSet/binaries. Since the packaging is 'pom', the project is NOT included in the moduleSet, because POMs are assumed not to have binaries. Or, if they do, they're assumed to have binaries that are addressable via some classifier...which I'm still not sure would even work with the assembly plugin.

Bottom line is, we need a better solution than abusing the pom packaging for assembly modules.

---

**Affects:** 2.2.1

**Issue Links:**
- [MNG-1683](https://issues.apache.org/jira/browse/MNG-1683) type zip for packaging ?

2 votes, 4 watchers
